staging/src/k8s.io/apiserver/pkg/util/flowcontrol/request/config.go

```diff
@@ -63,7 +65,13 @@ type MutatingWorkEstimatorConfig struct {
 }
 
 // DefaultWorkEstimatorConfig creates a new WorkEstimatorConfig with default values.
-func DefaultWorkEstimatorConfig() *WorkEstimatorConfig {
+func DefaultWorkEstimatorConfig(serverConcurrencyLimit int) *WorkEstimatorConfig {
+	// By default maximumSeats is 1% of total max inflight requests with an upper bound of 10
+	maximumSeats := uint64(math.Ceil((float64(serverConcurrencyLimit) * maximumSeatsFractionOfTotal)))
```

Comment:
While I understand the motivation behind this change (and I guess I agree with it), I'm not convinced the change as of now will not bring more damage then gain...

1. We are running smallest control-planes with max-inflights < 100 - this change automatically means that every single request will occupy 1 seat - is that really what we want?

2. We run large control-planes (in particular scalability tests) with < 1000 inflights, so we will never reach 10 max-seats for them. I'm fairly sure it's not really what we want.


So I guess I have two main concerns about this change:
1. `1%` seems too low to me
2. A second question is whether we want this to set that based on total max-seats, or rather per-priority-level (where per-priority-level in the borrowing world I rather mean sth like "max possible in-flights for a given priority assuming I borrow everything I can" (or sth like that - so that it doesn't change frequently).

@MikeSpreitzer

Reply:
> Actually, could we run this scalability test on this PR? Yes, max seats will never reach 10, but it will still be single digit value based on what is used in scalability tests, I'm curious if the impact will be noticeable if max seats is reduced from 10 to 7 (as an example).

Unfortunately, scalability test is not doing a good enough job in saturating control-plane so i don't think we will have useful data from it...


Now thinking more about my concerns:

1. Let's say someone sets inflight to 50 - then 4% has exactly the same problem. And if we bump it too high, then it will become uneffective.
So maybe my concerns boils down to the fact that a single value is not what we want.
We may need a function that takes `total inflights` as a parameter and returns a percent?

For low numbers it returns something higher, for higher numbers is decreases.
That said, I don't have a good suggestion how to create this function in more data-driven way...

> But for example sake, let's say we have a small control plane with --max-requests-inflight=100 and 500 pods

That's unclear to me - listing 500 pods is quite a bit of work too.
But I can definitely imagine that this value:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/util/flowcontrol/request/config.go#L28
when we run better tests, is just too low...

So yet another option would be to also compute this number based on `max-requests-inflight`.


2. For that, I agree with Mike that my concern is much more related to the issue about the fact that max-width can exceed the allocated capacity of one priority level.
I guess I'm fine of having that discussion separately from this PR.